test_conformance/basic/test_imagedim.cpp

```diff
@@ -41,8 +41,13 @@ static const char *image_dim_kernel_code =
 static unsigned char *
 generate_8888_image(int w, int h, MTdata d)
 {
-    unsigned char   *ptr = (unsigned char*)malloc(w * h * 4);
-    int             i;
+    unsigned char *ptr =
+        (unsigned char *)malloc(sizeof(unsigned char) * 4 * w * h);
+    unsigned int i;
+    if (ptr == NULL)
```

Comment:
I think we should NOT try to handle allocation failures by testing for NULL pointers, this will clutter the code for no real benefit. I suggest you change this to use `new[]` or a smart pointer.

Reply:
Addressed.